ConsoleGraphics/ConsoleProgressBar.cs

```diff
@@ -111,15 +158,32 @@ private void WriteRemainingProgressBlocks(uint completeProgressBlocks)
             }
         }
 
-        private void WriteProgressText(uint currentUnitOfWork, float percentComplete, ConsoleColor originalBackgroundColor)
+        private void WriteProgressText(uint currentUnitOfWork, double percentComplete, ConsoleColor originalBackgroundColor)
         {
             Console.BackgroundColor = originalBackgroundColor;
             Console.Write(" {0,5}% ({1} of {2})", percentComplete.ToString("n2"), currentUnitOfWork, TotalUnitsOfWork);
         }
 
+        private void WriteETA(TimeSpan eta, ConsoleColor originalBackgroundColor)
+        {
+            Console.BackgroundColor = originalBackgroundColor;
+            
+            if (eta < TimeSpan.FromMinutes(1))
+            {
+                Console.Write(" Less than a minute");
+            }
+            else
+            {
+                Console.Write(" {0} hour(s) {1} minute(s)", Math.Floor(eta.TotalHours), eta.Minutes);
```

Comment:
TimeSpan formatting would probably be better than assuming scale is hours and minutes.

Reply:
Yes, this can definitely be improved.     Consider my use case is on task of around 1 or 2 hours... not more...     I doubt there should be a Console app with a kind of foreground job that last more than X hours...      and worst case it will show something like   120 Day(s)...  

On my fork (not yet included here) i've added a check whereby if the time missing is less than a minute instead of showing 0 hour(s) 0 minute(s) shows "less than a minute"... 

Let me know what you prefer!

P.s. would be nice to generate even a Nuget package so that the library will be easy to use!